src/shogun/util/factory.h

```diff
@@ -99,7 +100,8 @@ namespace shogun
 	template <class T>
 	std::shared_ptr<Features> features(SGMatrix<T> mat)
```

Comment:
we might wanna move this to details namespace? @gf712

Reply:
++

Reply:
yes, i will be change those to details namespace when all refactor stuff is done, otherwise, there are too many errors :)